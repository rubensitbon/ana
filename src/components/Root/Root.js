// @flow
import React from 'react';
import { IntlProvider, addLocaleData } from 'react-intl';
import fr from 'react-intl/locale-data/fr';
import en from 'react-intl/locale-data/en';

import frMessages from 'translations/fr.json';
import enMessages from 'translations/en.json';
import { flattenMessages } from 'services/i18n/intl';
import logo from './logo.svg';
import StyledRoot from './Root.style';

const locales = {
  fr: flattenMessages(frMessages),
  en: flattenMessages(enMessages),
};

addLocaleData([...fr, ...en]);

type Props = {
  children: React$Element<*>,
};

export default class Root extends React.Component<Props> {
  render() {
    return (
      <IntlProvider locale="fr" messages={locales.fr}>
        <StyledRoot>
          <div className="header">
            <img src={logo} className="logo" alt="logo" />
            <h2>Welcome to React</h2>
          </div>
          {this.props.children}
        </StyledRoot>
      </IntlProvider>
    );
  }
}
