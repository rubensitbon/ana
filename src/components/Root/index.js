// @flow
export { default } from './Root.wrap';
