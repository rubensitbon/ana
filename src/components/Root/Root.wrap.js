import { connect } from 'react-redux';
import Root from './Root';

function mapStateToProps() {
  return {};
}

function mapDispatchToProps() {
  return {};
}

export default connect(mapStateToProps, mapDispatchToProps)(Root);
