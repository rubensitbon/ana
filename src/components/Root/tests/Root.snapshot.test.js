import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import Root from '../Root';

describe('<Root>', () => {
  it('should render correctly', () => {
    const props = {
      children: (
        <div>
          <span>test</span>
        </div>
      ),
    };
    const tree = shallow(<Root {...props} />);
    expect(toJSON(tree)).toMatchSnapshot();
  });
});
