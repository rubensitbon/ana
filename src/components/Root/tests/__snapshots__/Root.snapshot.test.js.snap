// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Root> should render correctly 1`] = `
<IntlProvider
  locale="fr"
  messages={
    Object {
      "home.click-me": "Click sur moi",
      "home.create-route": "Pour créer une route, modifie routes.js et associe une page",
      "home.generate-component": "Pour créer une page ou un simple composant, lance: ",
      "home.generate-module": "Pour ajouter de la logique métier, créer un dossier dans redux avec tes reducers/actions/sagas/selectors/constants/...",
      "home.get-started": "Pour commencer, modifie",
      "home.readme": "Lis le README.md pour découvrir les super commandes npm que nous avons ajouté",
      "home.save-to-reload": "et sauvegarde pour recharger.",
      "home.use-a-link": "Ou utilise un lien",
      "page.add-github-username": "Entrer un username github",
      "page.api-to-translate-example": "Ce message est traduit en utilisant directement l'API react-intl",
      "page.back": "Retour",
      "page.fetch-github-avatar": "Affiche l'avatar github",
    }
  }
>
  <styled.div>
    <div
      className="header"
    >
      <img
        alt="logo"
        className="logo"
        src="logo.svg"
      />
      <h2>
        Welcome to React
      </h2>
    </div>
    <div>
      <span>
        test
      </span>
    </div>
  </styled.div>
</IntlProvider>
`;
